open-api/rest-catalog-open-api.yaml

```diff
@@ -541,6 +541,263 @@ paths:
         5XX:
           $ref: '#/components/responses/ServerErrorResponse'
 
+  /v1/{prefix}/namespaces/{namespace}/tables/{table}/plan:
+    parameters:
+      - $ref: '#/components/parameters/prefix'
+      - $ref: '#/components/parameters/namespace'
+      - $ref: '#/components/parameters/table'
+    post:
+      tags:
+        - Catalog API
+      summary: Submit a scan for planning
+      description:
+        Submits a scan for server-side planning.
+
+
+        Point-in-time scans are planned by passing snapshot-id to identify the
+        table snapshot to scan. Incremental scans are planned by passing both
+        start-snapshot-id and end-snapshot-id. Requests that include both point
+        in time config properties and incremental config properties are
+        invalid. If the request does not include either incremental or
+        point-in-time config properties, scan planning should produce a
+        point-in-time scan of the latest snapshot in the table's main branch.
+
+
+        Responses must include a valid status
+
+        - When "completed" the planning operation has produced plan-tasks and
```

Comment:
> the planning operation has produced plan-tasks and file-scan-tasks that must be returned in the response

@rdblue I think I still wanted to confirm on this statement, as I do not think it is a requirement that the client "must" return both these types of tasks? I thought that its optional for the client to return only plan tasks, return only file scan tasks, or return both.

> two types of tasks (file scan tasks and plan tasks) and both may be included in the      response.

In `L580` we use this statement which uses "may" instead of "must" which seems to be more in line with my understanding, so assuming the first sentence should also match this?

Reply:
I updated this to specifically state the intent here is to say that they must be sent in the response, rather than fetched later. I think the wording below is clear already because it says that both lists are part of the response and both may be included.

This is further clarified by the structure of the response schemas.